search/views.py

```diff
@@ -0,0 +1,31 @@
+"""View for search"""
+from rest_framework.response import Response
+from rest_framework.status import HTTP_405_METHOD_NOT_ALLOWED
+from rest_framework.views import APIView
+
+from channels.constants import (
+    COMMENT_TYPE,
+    POST_TYPE,
+)
+from open_discussions import features
+from search.api import execute_search
+
+
+VALID_CONTENT_TYPES = ('', COMMENT_TYPE, POST_TYPE)
+VALID_SORT_OPTIONS = ('', 'score', 'num_comments', 'created')
+
+
+class SearchView(APIView):
+    """
+    View for executing searches
+    """
+    permission_classes = ()
+
+    def post(self, request, *args, **kwargs):
+        """Execute a search. Despite being POST this should not modify any data."""
+
+        if not features.is_enabled(features.SEARCH_UI):
+            return Response(status=HTTP_405_METHOD_NOT_ALLOWED)
+
+        response = execute_search(user=request.user, query=request.data)
```

Comment:
Any thoughts on how we should handle ES errors?

Reply:
I think an exception would get raised in the Elasticsearch library and it would become a 500 error. Do you think that's enough?

Reply:
I feel like if it's a `400 BAD REQUEST` from the ES backend, that should also be a 400 for our API since we're putting the ownership of the query on the frontend - a bad/invalid query shouldn't result in a 500.

Reply:
That's a good point